instrumentor/controllers/sourceinstrumentation/common.go

```diff
@@ -117,6 +118,110 @@ func syncNamespaceWorkloads(
 	return collectiveRes, errs
 }
 
+// syncRegexSourceWorkloads syncs all workloads that match the regex pattern in the source
+func syncRegexSourceWorkloads(
+	ctx context.Context,
+	k8sClient client.Client,
+	runtimeScheme *runtime.Scheme,
+	source *odigosv1.Source) (ctrl.Result, error) {
+
+	pattern := source.Spec.Workload.Name
+	namespace := source.Spec.Workload.Namespace
+	kind := source.Spec.Workload.Kind
+
+	// Compile the regex pattern
+	regex, err := regexp.Compile(pattern)
+	if err != nil {
+		// Invalid regex pattern, return error
+		return ctrl.Result{}, err
+	}
+
+	workloadsToSync := make([]k8sconsts.PodWorkload, 0)
+	collectiveRes := ctrl.Result{}
+	var errs error
+
+	// List all workloads of the specified kind in the namespace
+	workloadObjects := workload.ClientListObjectFromWorkloadKind(kind)
+	err = k8sClient.List(ctx, workloadObjects, client.InNamespace(namespace))
+	if err != nil {
+		// Ignore "no matches for kind" and "forbidden" errors for DeploymentConfig
```

Comment:
can you elaborate more on the reason to do this exception - in what situation we can run into these errors? and why only DeploymentConfig specifically